test/search_api/endpoints/test_get_entity_by_pid.py

```diff
@@ -0,0 +1,391 @@
+                            "modId": "user",
+                            "modTime": "2005-04-30 19:41:49+00:00",
+                            "role": "CI",
+                            "createTime": "2002-11-27 06:20:36+00:00",
+                            "id": 1,
+                        },
+                    ],
+                    "endDate": "2000-07-09 00:00:00+00:00",
+                    "modId": "user",
+                    "title": "Including spend increase ability music skill former."
+                    " Agreement director concern once technology sometimes someone"
+                    " staff.\nSuccess pull bar. Laugh senior example.",
+                    "summary": "Season identify professor happen third. Beat"
+                    " professional blue clear style have. Light final summer.",
+                    "visitId": "42",
+                    "parameters": [
+                        {
+                            "dateTimeValue": "2000-05-07 00:00:00+00:00",
+                            "rangeBottom": 48.0,
+                            "createId": "user",
+                            "numericValue": 127265.0,
+                            "modTime": "2005-04-30 19:41:49+00:00",
+                            "createTime": "2002-11-27 06:20:36+00:00",
+                            "id": 1,
+                            "stringValue": "international1",
+                            "modId": "user",
+                            "rangeTop": 101.0,
+                            "error": 31472.0,
+                        },
+                    ],
+                    "name": "INVESTIGATION 1",
+                    "releaseDate": "2000-07-05 00:00:00+00:00",
+                    "id": 1,
+                },
+                id="Get document by pid including all possible related entities",
+                # Skipped because of incorrect members JSON naming
+                # (investigationUsers key used instead). It's in ICAT fields too,
+                # not converted into PaNOSC format
+                marks=pytest.mark.skip,
+            ),
+        ],
+    )
+    def test_valid_get_by_pid_endpoint(
+        self,
+        flask_test_app_search_api,
+        endpoint_name,
+        pid,
+        request_filter,
+        expected_json,
+    ):
+        test_response = flask_test_app_search_api.get(
+            f"{Config.config.search_api.extension}/{endpoint_name}/{pid}?filter="
+            f"{request_filter}",
+        )
+
+        assert test_response.status_code == 200
+        assert test_response.json == expected_json
+
+    @pytest.mark.parametrize(
+        "pid, request_filter, expected_status_code",
+        [
+            pytest.param("0-8401-1070-7", '{"where": []}', 400, id="Bad where filter"),
+            pytest.param("0-8401-1070-7", '{"limit": -1}', 400, id="Bad limit filter"),
+            pytest.param(
+                "0-8401-1070-7", '{"limit": -100}', 400, id="Bad skip filter",
```

Comment:
This should be `skip`
```suggestion
                "0-8401-1070-7", '{"skip": -100}', 400, id="Bad skip filter",
```